Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/DefaultPrimaryPointerSelector.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Default primary pointer selector. The primary pointer is chosen among all interaction enabled ones using the following rules in order:
+    ///   1. Currently pressed pointer that has been pressed for the longest
+    ///   2. Pointer that was released most recently
+    ///   3. Pointer that became interaction enabled most recently
+    /// </summary>
+    public class DefaultPrimaryPointerSelector : IMixedRealityPrimaryPointerSelector
+    {
+        private readonly Dictionary<IMixedRealityPointer, PointerInfo> pointerInfos = new Dictionary<IMixedRealityPointer, PointerInfo>();
+
+        public void Initialize()
+        {
+            if (MixedRealityToolkit.InputSystem != null)
+            {
+                MixedRealityToolkit.InputSystem.PointerEvent += OnPointerEvent;
+            }
+        }
+
+        public void Destroy()
+        {
+            if (MixedRealityToolkit.InputSystem != null)
+            {
+                MixedRealityToolkit.InputSystem.PointerEvent -= OnPointerEvent;
+            }
+        }
+
+        #region IMixedRealityPrimaryPointerSelector
+
+        public void RegisterPointer(IMixedRealityPointer pointer)
+        {
+            var pointerInfo = new PointerInfo(pointer);
+            pointerInfos.Add(pointer, pointerInfo);
+        }
+
+        public void UnregisterPointer(IMixedRealityPointer pointer)
+        {
+            pointerInfos.Remove(pointer);
+        }
+
+        public IMixedRealityPointer Update()
+        {
+            IMixedRealityPointer primaryPointer = null;
+            PointerInfo primaryInfo = null;
+
+            foreach (var keyValue in pointerInfos)
+            {
+                var pointer = keyValue.Key;
+                var info = keyValue.Value;
+                info.Update(pointer);
+
+                if (info.IsInteractionEnabled && 
```

Comment:
It would be fantastic to be able to filter out pointers that are not near interaction pointers.
In most cases, the primary pointer is for interacting with UI, and so it must be a far interaction pointer, especially when driving legacy UIs

Reply:
Ideally, I would like to have a profile for the default selector with common configuration options. I have very limited time to work on this though, so for the time being I'm going to make DefaultPrimaryPointerSelector.Update virtual so you can implement that behavior easily yourself.

Reply:
This is a fine solution.